patches/0001-Expose-Encoder.Canonical-and-MarshalCanonical-Handle.patch

```diff
@@ -380,21 +377,15 @@ index fca2a09..aaa79c2 100644
 +	} else {
 +		x.byIndex = f
 +	}
++	newM[t] = x
+ 	fieldCache.value.Store(newM)
```

Comment:
@tiborvass found that The previous code did not actually do anything with `x` - it did not set it back to `fieldCache`, which was a bug.  When encoding JSON, it seems like some fields didn't actually get sorted by name.

So we ended up with serialized values like:  `{"keytype":"ecdsa-x509","keyval":{"private":null,"public":"..."}}` where the keys in `"keyval"` were not lexicographically ordered.

The updated code does set `x` back into the `fieldCache`, so we end up with items like: `{"keytype":"ecdsa-x509","keyval":{"public":"...","private":null}}`, which is correctly lexicographically ordered.

This may cause backwards compatibility issues in users of previous versions of canonical JSON.   cc @endophage @riyazdf